test_volume.py

```diff
@@ -178,3 +179,198 @@ def test_maximum_number_of_volumes(server, create_volume):
     assert error.value.response.json()['detail'] == (
         "Due to internal limitations, it is currently not possible "
         "to attach more than 128 volumes.")
+
+
+def test_snapshot_volume_attached(server, volume):
+    """ Attached volumes can be snapshotted and reverted.
+
+    It is possible to create a snapshot of a volume which is currently attached
+    and to revert the volume back to this state.
+
+    Snapshots of volumes taken while they are attached and mounted are crash
+    consistent. Some in-flight data might not be in the snapshot, but volume
+    can always be recovered to a consistent state.
+    """
+
+    # Attach and format volume to server
+    volume.attach(server)
+    time.sleep(5)
+    server.assert_run('sudo mkfs.ext4 /dev/sdb')
+    server.assert_run('sudo mount /dev/sdb /mnt')
+
+    # Create two files. The first is synced to disk with fsync, the second
+    # is not synced. Data might still be in-flight.
+    server.assert_run(
+        'sudo dd if=/dev/zero of=/mnt/synced count=1 bs=1M conv=fsync')
+    server.assert_run('sudo dd if=/dev/zero of=/mnt/not-synced count=1 bs=1M')
```

Comment:
For uniformity you might consider adding a newline here too, even though the line is not too longh.

Reply:
Done